BasiliskII/src/CrossPlatform/vm_alloc.cpp

```diff
@@ -255,9 +269,12 @@ void * vm_acquire(size_t size, int options)
 	if ((addr = mmap((caddr_t)next_address, size, VM_PAGE_DEFAULT, the_map_flags, fd, 0)) == (void *)MAP_FAILED)
 		return VM_MAP_FAILED;
 	
-	// Sanity checks for 64-bit platforms
+#if DIRECT_ADDRESSING
```

Comment:
I'm not clear on why this check should now happen _only_ for `DIRECT_ADDRESSING`, can you explain?  Did it crash on all 64-bit before this?

Reply:
It likely crashes on all 64-bit Linux other than x86_64 yes.

x86_64 only works because it is the only linux arch that implements the MAP_32BIT flag. All others ignore it. :)

You can confirm that in the source here: https://elixir.bootlin.com/linux/latest/ident/MAP_32BIT

Reply:
Okay I just spent a day trying to resolve the 64-bit issues in DIRECT_ADDRESSING and it's a quagmire. So the sanity check is definitely needed there.

I'll note it was rickyzhang who pointed out you don't need the check when banked addressing. :)

Reply:
I'd explicitly note the relation to `MAP_32BIT` a comment as well

Reply:
TBH I'm working on making all this brittle mmap stuff go away entirely. But this gets it working for now. :)